ethereum/contracts/relayer/coreRelayer/CoreRelayerDelivery.sol

```diff
@@ -1,623 +1,560 @@
+    uint256 transactionFeeRefundAmount,
+    bool receiverValueWasPaid,
+    uint256 transactionFeeRefundAmountPostForward
+  ) private returns (RefundStatus refundStatus) {
+    //Amount of receiverValue that is refunded to the user (0 if the call to
+    //  'receiveWormholeMessages' did not revert, or the full receiverValue otherwise)
+    uint256 receiverValueRefundAmount =
+      (receiverValueWasPaid ? 0 : deliveryInstruction.receiverValueTarget);
+
+    //Total refund to the user
+    uint256 refundToRefundAddress =
+      receiverValueRefundAmount + transactionFeeRefundAmountPostForward;
+    
+    //Refund the user
+    refundStatus = payRefundToRefundAddress(
+      deliveryInstruction.refundChainId,
+      deliveryInstruction.refundAddress,
+      refundToRefundAddress,
+      deliveryInstruction.targetRelayProvider
+    );
 
-    function processOverrides(
-        IWormholeRelayerInternalStructs.DeliveryInstruction memory deliveryInstruction,
-        bytes memory encoded
-    )
-        internal
-        pure
-        returns (
-            IWormholeRelayerInternalStructs.DeliveryInstruction memory withOverrides,
-            bytes32 redelivery
-        )
-    {
-        if (encoded.length == 0) {
-            return (deliveryInstruction, 0x0);
-        } else {
-            IDelivery.DeliveryOverride memory overrides = decodeDeliveryOverride(encoded);
-            if (overrides.gasLimit < deliveryInstruction.executionParameters.gasLimit) {
-                revert IDelivery.InvalidOverrideGasLimit();
-            } else if (overrides.receiverValue < deliveryInstruction.receiverValueTarget) {
-                revert IDelivery.InvalidOverrideReceiverValue();
-            } else if (overrides.maximumRefund < deliveryInstruction.maximumRefundTarget) {
-                revert IDelivery.InvalidOverrideMaximumRefund();
-            }
-
-            deliveryInstruction.executionParameters.gasLimit = overrides.gasLimit;
-            deliveryInstruction.receiverValueTarget = overrides.receiverValue;
-            deliveryInstruction.maximumRefundTarget = overrides.maximumRefund;
-            return (deliveryInstruction, overrides.redeliveryHash);
-        }
+    //Refund the relayer (their extra funds) + (the amount that the relayer spent on gas)
+    //  + (the users refund if that refund didn't succeed)
+    uint256 relayerRefundAmount = (
+      msg.value - deliveryInstruction.receiverValueTarget - deliveryInstruction.maximumRefundTarget
+    ) + (deliveryInstruction.maximumRefundTarget - transactionFeeRefundAmount)
+    //TODO AMO: Isn't this a bug? We add the same amount regardless of whether we hit the max or not
+      + ((refundStatus == RefundStatus.REFUND_SENT ||
+          refundStatus == RefundStatus.CROSS_CHAIN_REFUND_SENT ||
+          refundStatus == RefundStatus.CROSS_CHAIN_REFUND_SENT_MAXIMUM_BUDGET
+         ) ? 0 : refundToRefundAddress);
+
+    //TODO AMO: what if pay fails? (i.e. returns false)
```

Comment:
If it fails, I guess it just gets stored in the contract. What would you suggest we do in that case?

Reply:
As I said for the other pay too: We probably want to enforce that `pay` is successful because (unless I'm missing something) this is fully under the control of the relayProvider. Seems kinda bad if we don't "warn" them (via a failing transaction simulation) that they won't get rewarded for their relay because their setup is somehow screwed up.